libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,202 @@
+  private static final String NEW_LINE = "\n";
+  private static final String QUESTION_MARK = "?";
+  private static final String EQAL_SIGN = "=";
+  private static final int BLACKLIST_HEAD = 0;
+  private static final String REVOKED_CERT_KEYS = "RevokedCertKeys";
+  private static final String COM_CONFIG_ENDPOINT = "https://api.mapbox.com/events-config";
+  private static final String CHINA_CONFIG_ENDPOINT = "https://api.mapbox.cn/events-config";
+  private static final String ACCESS_TOKEN_QUERY_PARAMETER = "access_token";
+  private static final String EXTRACTION_EXEMPTION = "Blacklist Response Extraction Failed";
+  private static final String ENDPOINT_DETERMINATION_FAIL = "Endpoint Determination Failed";
+  private static final String SAVE_BLACKLIST_FAIL = "Unable to save blacklist to file";
+  private static final String RETRIEVE_TIME_FAIL = "Unable to retrieve last update time from blacklist";
+  private static final String RETRIEVE_BLACKLIST_FAIL = "Unable to retrieve blacklist contents from file";
+  private static final String REQUEST_FAIL = "Request failed to download blacklist";
+  private static final Map<Environment, String> ENDPOINTS = new HashMap<Environment, String>() {
+    {
+      put(Environment.COM, COM_CONFIG_ENDPOINT);
+      put(Environment.CHINA, CHINA_CONFIG_ENDPOINT);
+    }
+  };
+  private Context context;
+  private String accessToken;
+
+  CertificateBlacklist(Context context, String accessToken) {
+    this.context = context;
+    this.accessToken = accessToken;
+  }
+
+  List<String> retrieveBlackList() {
+    File directory = context.getFilesDir();
+    List<String> blacklist = new ArrayList<>();
+
+    if (directory.isDirectory()) {
+      File file = new File(directory, BLACKLIST_FILE);
+
+      if (file.exists()) {
+        try {
+          blacklist = obtainBlacklistContents(file);
+          blacklist.remove(BLACKLIST_HEAD);
+        } catch (IOException exception) {
+          Log.e(LOG_TAG, RETRIEVE_BLACKLIST_FAIL, exception);
+        }
+      }
+    }
+
+    return blacklist;
+  }
+
+  boolean daySinceLastUpdate() {
+    long millisecondDiff = System.currentTimeMillis() - retrieveLastUpdateTime();
+    long dayInMilliseconds = 86400000;
+
+    return millisecondDiff >= dayInMilliseconds;
+  }
+
+  private long retrieveLastUpdateTime() {
+    File directory = context.getFilesDir();
+    File file = new File(directory, BLACKLIST_FILE);
+
+    long lastUpdateTime = 0;
+
+    if (file.exists()) {
+      try {
+        List<String> blacklist = obtainBlacklistContents(file);
+        lastUpdateTime = Long.valueOf(blacklist.get(BLACKLIST_HEAD));
+      } catch (IOException exception) {
+        Log.e(LOG_TAG, RETRIEVE_TIME_FAIL, exception);
+      }
+    }
+
+    return lastUpdateTime;
+  }
+
+  void updateBlacklist() {
+    Request request = new Request.Builder()
+      .url(determineConfigEndpoint() + QUESTION_MARK + ACCESS_TOKEN_QUERY_PARAMETER + EQAL_SIGN + accessToken)
```

Comment:
As commented in https://github.com/mapbox/mapbox-events-android/pull/151#discussion_r202348319 what about 👇

> passing a `HttpUrl` instead (we're currently doing that in `TelemetryClient` for example)? That way we'd able to add query parameters easily 👀 https://square.github.io/okhttp/3.x/okhttp/okhttp3/HttpUrl.Builder.html#addQueryParameter-java.lang.String-java.lang.String-